python/lsst/afw/image/slicing.py

```diff
@@ -31,9 +31,30 @@ def _getBBoxFromSliceTuple(img, imageSlice):
      im[-2, -2]
      im[1:4, 6:10]
      im[:]
+
+    You may also add an extra argument, afwImage.PARENT or afwImage.LOCAL.  The default is LOCAL
+    as before, but if you specify PARENT the bounding box is interpreted in PARENT coordinates
+    (this includes slices; e.g.
+     im[-3:, -2:, afwImage.PARENT]
+    still means the last few rows and columns, and 
+     im[1001:1004, 2006:2010, afwImage.PARENT]
+    with xy0 = (1000, 2000) refers to the same pixels as
+     im[1:4, 6:10, afwImage.LOCAL]
+    )
     """
+    origin = LOCAL                      # this sets the default value
+
+    try:
+        _origin = imageSlice[-1]
+    except TypeError:
+        _origin = None
+
+    if isinstance(_origin, ImageOrigin):
+        origin = _origin
+        imageSlice = imageSlice[0] if len(imageSlice) <= 2 else imageSlice[:-1]
```

Comment:
I must be missing something here too.  It seems like imageSlice would only ever be 2 or 3 elements long.  If it's 3, it has an origin, if not you want both the elements.  Why only pass on the first element?